app/tests/reader_studies_tests/test_api.py

```diff
@@ -1657,6 +1658,88 @@ def test_display_set_add_and_edit(client, settings):
     )
 
 
+@pytest.mark.django_db
+def test_display_set_index(client):
+    rs = ReaderStudyFactory()
+    r = UserFactory()
+    rs.add_reader(r)
+
+    for _ in range(20):
+        civ = ComponentInterfaceValueFactory(image=ImageFactory())
+        ds = DisplaySetFactory(reader_study=rs)
+        ds.values.add(civ)
```

Comment:
This is going to be quite slow. Can you do this with less objects? Also remember about the create/build batch methods of factory boy. You might be able to hardcode the users pk for seed setting to make this deterministic.